providers/aws-ec2/src/main/java/org/jclouds/aws/ec2/compute/loaders/AWSEC2CreateSecurityGroupIfNeeded.java

```diff
@@ -102,7 +104,17 @@ private String createSecurityGroupInRegion(String region, String name, String vp
                                .build());
             }
 
-            String myOwnerId = Iterables.get(securityApi.describeSecurityGroupsInRegion(region, name), 0).getOwnerId();
+            // Use filter (as per `SecurityGroupPresent`, in securityGroupEventualConsistencyDelay)
+            Set<SecurityGroup> securityGroups = securityApi.describeSecurityGroupsInRegionWithFilter(region,
+                  ImmutableMultimap.of("group-name", name));
```

Comment:
You could use the `id` of the security group with a `group-id` filter here. Same for `securityGroupEventualConsistencyDelay`, by passing `new RegionAndName(region, id)`. The ID is unique for the region, while the name can be the same between VPCs - will guarantee that you don't get multiple results here.